src/jit/inline.cpp

```diff
@@ -697,7 +697,9 @@ void InlineResult::Report()
 
         InlineObservation obs = m_Policy->GetObservation();
 
-        if ((m_Callee != nullptr) && (obs != InlineObservation::CALLEE_IS_NOINLINE))
+        if ((m_Callee != nullptr) && 
+            (obs != InlineObservation::CALLEE_IS_NOINLINE) && 
+            (obs != InlineObservation::CALLEE_DOES_NOT_RETURN))
```

Comment:
> It would be good noet that here and link to the issue you opened for recording noinline deductions with the runtime.

You mean the issue you opened and I hijacked? :) - #6095